app/src/main/java/com/example/vshcheglov/webshop/presentation/main/MainViewModel.kt

```diff
@@ -123,20 +133,22 @@ class MainViewModel : ViewModel() {
 
     }
 
-    //This Method called from OnActivityResult (before onResume) => view == null
-    fun updateUserProfilePhoto(profilePhotoBitmap: Bitmap) {
-        _stateLiveData.value = getState().copy(avatarImage = profilePhotoBitmap)
-        saveUserAvatar()
-    }
+    private fun saveUserAvatar(imagePath: String) {
+        val constraints = Constraints.Builder()
```

Comment:
I would extract these block of code to separate class to encapsulate `WorkManager ` calls there